controller_interface/include/semantic_components/semantic_component_interface.hpp

```diff
@@ -85,7 +91,7 @@ class SemanticComponentInterface
       return false;
     }
     // insert all the values
-    for (size_t i = 0; i < state_interfaces_.size(); ++i)
+    for (auto i = 0; i < state_interfaces_.size(); ++i)
```

Comment:
```
/workspaces/ros2_rolling_ws/src/ros2_control/controller_interface/include/semantic_components/semantic_component_interface.hpp:96:45: error: implicit conversion changes signedness: 'int' to 'size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
   96 |       values.emplace_back(state_interfaces_[i].get().get_value());
      |                           ~~~~~~~~~~~~~~~~~ ^
/workspaces/ros2_rolling_ws/src/ros2_control/controller_interface/include/semantic_components/semantic_component_interface.hpp:94:24: warning: comparison of integers of different signs: 'int' and 'size_type' (aka 'unsigned long') [-Wsign-compare]
   94 |     for (auto i = 0; i < state_interfaces_.size(); ++i)
      |                      ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~
```

```suggestion
    for (size_t i = 0; i < state_interfaces_.size(); ++i)
```

Reply:
We agreed to use auto, I added u literal to avoid warning.